modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/ObjectFieldFormBase.tsx

```diff
@@ -70,6 +70,16 @@ export type ObjectFieldErrors = FormError<
 const defaultLanguageId = Liferay.ThemeDisplay.getDefaultLanguageId();
 
 const fieldSettingsMap = new Map<string, ObjectFieldSetting[]>([
+	[
+		'Aggregation',
+		[
+			{
+				name: 'filters',
+				objectFieldId: 0,
+				value: Array(0),
+			},
+		],
+	],
```

Comment:
Hey @manuelecastro, good job on that! Your solution is good! Just a suggestion, if you notice in the browser console, we are having errors even with your solution, because the code is still trying to make some request with the `objectDefinitionExternalReferenceCode2` which in this specific behavior we still don't have at the moment when the user changes the field type . So, to avoid possible bugs because of this and to clear the browser console of errors, we can cover those cases as well.

Then in `AggregationFilterContainer.tsx` in the `useEffect` line `108` you can cover all function content by a condition, checking if `objectDefinitionExternalReferenceCode2` exists like:

```tsx
useEffect(() => {
    if(objectDefinitionExternalReferenceCode2) {
	const makeFetch = async () => {
	    const items = await API.getObjectFieldsByExternalReferenceCode(
		objectDefinitionExternalReferenceCode2!
	    );

	    const objectDefinition2 = await API.getObjectDefinitionByExternalReferenceCode(
		objectDefinitionExternalReferenceCode2!
	    );
	    setCreationLanguageId2(objectDefinition2.defaultLanguageId);

	    setObjectFields(items);
	};

	makeFetch();
}
}, [objectDefinitionExternalReferenceCode2, setCreationLanguageId2]);
```

The other place is `AggregationFormBase.tsx ` in the `useEffect` line `145`, here you can cover everything including the `relatedFields` by a condition checking if the `currentRelatedObjectRelationship` exists, like:

```tsx

useEffect(() => {
    if (editingField && objectRelationships) {
	const makeFetch = async () => {
		const settings = normalizeFieldSettings(objectFieldSettings);

		const currentRelatedObjectRelationship = objectRelationships.find(
			(relationship) =>
				relationship.name === settings.objectRelationshipName
		) as ObjectRelationship;

		const currentFunction = aggregationFunctions.find(
			(aggregationFunction) =>
				aggregationFunction.value === settings.function
		);

		if(currentRelatedObjectRelationship) {					
			const relatedFields = await API.getObjectFieldsByExternalReferenceCode(
				currentRelatedObjectRelationship.objectDefinitionExternalReferenceCode2
			);

			const currentSummarizeField = relatedFields.find(
				(relatedField) =>
					relatedField.name === settings.objectFieldName
			) as ObjectField;

			if (onRelationshipChange) {
				onRelationshipChange(
					currentRelatedObjectRelationship.objectDefinitionExternalReferenceCode2
				);
			}

			setObjectRelationshipFields(
				relatedFields.filter(
					(objectField) =>
						objectField.businessType === 'Integer' ||
						objectField.businessType === 'LongInteger' ||
						objectField.businessType === 'Decimal' ||
						objectField.businessType === 'PrecisionDecimal'
				)
			);

			setSelectRelatedObjectRelationship(
				currentRelatedObjectRelationship
			);

			setSelectedAggregationFunction(currentFunction);

			if (currentSummarizeField) {
				setSelectedSummarizeField(
					getLocalizableLabel(
						creationLanguageId2 as Liferay.Language.Locale,
						currentSummarizeField.label,
						currentSummarizeField.name
					)
				);
			}
		}
	};

	makeFetch();
    }
}, [
    creationLanguageId2,
    editingField,
    objectRelationships,
    objectFieldSettings,
    onRelationshipChange,
]);
```